workflow/snakemake_rules/common.smk

```diff
@@ -145,15 +145,37 @@ def _get_max_date_for_frequencies(wildcards):
             date.today() - offset
         )
 
-def _get_first(config, *keys):
-    """
-    Get the value of the first key in *keys* that exists in *config* and has a
-    non-empty value.
-
-    Raises a :class:`KeyError` if none of the *keys* exist with a suitable
-    value.
-    """
-    for key in keys:
-        if config.get(key) not in {"", None}:
-            return config[key]
-    raise KeyError(str(keys))
+def _get_upload_inputs(wildcards):
```

Comment:
As the uploads are only intended for internal use (they're only used by the `export_for_nextstrain` rule) would they be better co-located in ` workflow/snakemake_rules/export_for_nextstrain.smk`?

Reply:
I do think they're only for internal use, and my first instinct was to put this `_get_upload_inputs` function alongside the `export_for_nextstrain` rule. But looking more widely, I saw that all input functions seemed to be colocated in `common.smk`, even when they're tightly coupled to a single caller, so I wanted to follow that convention. Happy to change that convention if we want.